src/shared/IndexedDBConfigCache.ts

```diff
@@ -0,0 +1,69 @@
+import type { IConfigCache, IConfigCatCache } from "../ConfigCatCache";
+import { ExternalConfigCache } from "../ConfigCatCache";
+import type { OptionsBase } from "../ConfigCatClientOptions";
+
+const OBJECT_STORE_NAME = "configCache";
+
+type DBConnectionFactory = () => Promise<IDBDatabase>;
+
+export class IndexedDBConfigCache implements IConfigCatCache {
+  static tryGetFactory(): ((options: OptionsBase) => IConfigCache) | undefined {
+    const dbConnectionFactory = getDBConnectionFactory();
+    if (dbConnectionFactory) {
+      return options => new ExternalConfigCache(new IndexedDBConfigCache(dbConnectionFactory), options.logger);
+    }
+  }
+
+  constructor(private readonly dbConnectionFactory: DBConnectionFactory) {
+  }
+
+  async set(key: string, value: string): Promise<void> {
+    const db = await this.dbConnectionFactory();
+    try {
+      await new Promise<void>((resolve, reject) => {
+        const transaction = db.transaction(OBJECT_STORE_NAME, "readwrite");
+        const store = transaction.objectStore(OBJECT_STORE_NAME);
+        store.put(value, key);
+        transaction.oncomplete = () => resolve();
```

Comment:
i'm not 100% sure how the transaction works here, but maybe if the store.put completes before this line, the   transaction.oncomplete = () => resolve(); will not fire? Just a guess, but maybe it would be safer to put the transaction.oncomplete and onerror before the store.put